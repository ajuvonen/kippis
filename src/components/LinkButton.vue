<script setup lang="ts">
import {RouterLink} from 'vue-router';

defineProps<{
  to: string;
}>();
</script>
<template>
  <RouterLink class="link-button" :to="to">
    <slot />
  </RouterLink>
</template>
<style lang="scss" scoped>
.link-button {
  @apply flex items-center;
}
</style>
