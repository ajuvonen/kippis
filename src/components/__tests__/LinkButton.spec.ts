import {mount} from '@vue/test-utils';
import {describe, it, expect} from 'vitest';
import LinkButton from '@/components/LinkButton.vue';

describe('LinkButton', () => {
  it('mounts', () => {
    const wrapper = mount(LinkButton, {props: {to: '/'}, slots: {default: 'My Link'}});
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('points to search', () => {
    const wrapper = mount(LinkButton, {props: {to: '/search'}});
    expect(wrapper.find('a[href="/search"]').exists()).toBe(true);
  });
});
