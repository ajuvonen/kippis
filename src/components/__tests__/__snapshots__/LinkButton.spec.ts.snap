// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LinkButton > mounts 1`] = `"<a data-v-c2a26f37=\\"\\" href=\\"/\\" class=\\"link-button\\">My Link</a>"`;
